Create ui-c8y-1021-28-0-migrate-map-components-used-in-dm-and-remove-legacy-code-where.md (#2754)
SoftwareAG/c8y-docs: content/change-logs/device-management/ui-c8y-1021-28-0-migrate-map-components-used-in-dm-and-remove-legacy-code-where.md

---
date: ""

> title: Migrate map components used in DM and remove legacy code where possible

Author: The Claude AI suggested to use the following title:
```suggestion
title: Device management map components migrated and legacy code removed
```

> title: Migrate map components used in DM and remove legacy code where possible

Reviewer: ```suggestion
title: Migrate device map Device Management
```

Reviewer 2: ```suggestion
title: Migrated and improved the devices map in the Device Management application
```

product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-3089
version: 1021.28.0
---

> Migrate map components used in DM and remove legacy code where possible

Author: The Claude AI suggested to use the following body:
```suggestion
As part of ongoing improvements to device management, the map components used in the device management application have been migrated to a new implementation. The goal of this migration is to keep the map functionality up-to-date and maintainable. With this change, the map components in device management now use a modernized codebase. For users, the map functionality remains the same as before, so there should be no visible impact. As part of this migration, legacy code related to the previous map implementation has also been removed where possible, without affecting any functionality.
```

> Migrate map components used in DM and remove legacy code where possible

Reviewer: ```suggestion
As part of ongoing improvements to Device Management, the device map component (Devices > Map) has been migrated to a new implementation. Besides improving the functionality, the goal of this migration is to keep the component up-to-date and maintainable. The limit of 100 devices displayed has been removed. Areas with large density of devices are marked by clustering devices into a single marker displaying the number of devices in the area. World map is not wrapped and users can endlessly scroll to the east and to the west. Automatic initial zoom and center based on the determined device positions introduced. Auto-refresh with pause/resume functionality replaces realtime updates.
```

Reviewer 2: ```suggestion
As part of ongoing improvements to the Device Management application, the device map component (Devices > Map) has been migrated to a new implementation. Besides improving the functionality, the goal of this migration is to keep the component up-to-date and maintainable. The following changes have been made:

- The limit of 100 devices displayed has been removed. 
- Areas with large density of devices are marked by clustering devices into a single marker displaying the number of devices in the area. 
- The world map is not wrapped and users can endlessly scroll to the east and to the west. 
- Automatic initial zoom and center based on the determined device positions has been introduced. 
- Auto-refresh with pause/resume functionality replaces realtime updates.
```
